### DISCUSSION TITLE
Cant get into maskrom

### DISCUSSION THREAD

**Discussion by EnCodecwtf** [ntherning/StorytelReaderMods#3] [Q&A]
Cant get in to maskrom. New device gen2 the cable that follows.
Device showed upp for some seconds in devicemagment but then devisen loads into recovery mode and it drops från magment.

If plugin and plug out cable repetly it doesnt boot into recovery but when plug in cable it does.

**Comment by ntherning** [comment]
Hi!

Sounds like you are running Windows. Did you install the USB driver using Zadig as explained in the [Installing rkflashtool](https://github.com/ntherning/StorytelReaderMods#installing-rkflashtool) section?

Please provide a screenshot of Device Manager when your reader shows up.

**Reply by EnCodecwtf** [reply]
Thats right.
I actually manage to get into maskrom after installation of driver through Zadig.

But I have not been lucky to 100% Sadly.
I think i have installed all the software on the computer that I need.

And I have luckily backup of the boot.
Iam pretty sure I have installed the boot-grn2 patch successfuly.

And I managed to restart the system through rk flashtool.

But now Im stuck I think, course I cant install any apps. How do I install through abd in gen2
Tried to install through rkflashtool but it didnt work.

**Reply by EnCodecwtf** [reply]
After enabling maskrom ändå inställd the pathed boot.
Its impossible to load recovery mode.
Is it correct course I cant.

**Reply by ntherning** [reply]
After you have flashed the `boot-gen2-patched.img` file using `rkflashtool` to the `boot` partition you should reboot the device and if everything is ok it should boot into the usual Storytel app.

Can you please confirm that this is what happens when you boot the device? If not, what happens?

I have updated the Quickstart with a 7th step on how to get ADB: https://github.com/ntherning/StorytelReaderMods#quickstart.

**Reply by EnCodecwtf** [reply]
Thanks.
Thats right.
My question, is where i have to be in the Storytel device to use adb.
Is it in maskrom?
Is it in recovery mode?
Is it in the Storytel app?

Because in maskrom its not working.

**Reply by EnCodecwtf** [reply]
I finally got it to work I guess.
I gonna write down what I did later, and post here.
Maybe it is a help for someone

**Reply by ntherning** [reply]
Hi! I am very pleased to hear that you got it working! Please share your experience and I will update the instructions to help others. Thanks!

**Reply by RasmusAChr** [reply]
Hi, I had the exact same problem, but I got it to work!
I had around 10 seconds in maskrom before it would continue to recovery mode. I just had Zadig open, so when i could see the device on my pc, it would automatically select in Zadig, and when i clicked 'Install Driver' in never went back to recovery mode :)
I spent around 1 hour just restarting my reader :,)

**Reply by bharosM** [reply]
Can confirm, had the exact same problem, and worked in the same way. Just had to install WinUSB via Zadig fast enough. It told me it had failed, but when I rebooted the device again instead of going into recovery, it booted into maskrom.
